src/library/scala/collection/immutable/Range.scala

```diff
@@ -153,19 +153,15 @@ extends scala.collection.AbstractSeq[Int]
   }
 
   @inline final override def foreach[@specialized(Unit) U](f: Int => U) {
-    validateMaxLength()
```

Comment:
However, that test was lost. I'd be happy if it was deemed irrelevant, but I'd like to make sure this was a conscious decision.